plugin/pkg/shoot/validator/admission.go

```diff
@@ -214,8 +214,13 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 		}
 	}
 
-	// Check whether seed is protected or not. In case it is protected then we only allow Shoot resources to reference it which are part of the Garden namespace.
-	if shoot.Namespace != v1beta1constants.GardenNamespace && seed != nil && helper.TaintsHave(seed.Spec.Taints, garden.SeedTaintProtected) {
+	changed, err := seedChanged(a)
+	if err != nil {
+		return apierrors.NewInternalError(err)
+	}
+	needCheckForProtectedSeed := changed || a.GetOperation() == admission.Create
+	// Check whether seed is protected or not only if the shoot.spec.seed has been updated. In case it is protected then we only allow Shoot resources to reference it which are part of the Garden namespace.
```

Comment:
```suggestion
	// Check whether seed is protected or not only if the shoot.spec.seedName has been updated. In case it is protected then we only allow Shoot resources to reference it which are part of the Garden namespace.
```

Reply:
nice catch, thanks!